test/tests/mortar/continuity-3d-non-conforming/tests

```diff
@@ -36,6 +36,7 @@
                   'mortar constraint'
     min_ad_size = 200
     max_parallel = 1 # Serial needed for debugging mesh
+    heavy = True
```

Comment:
Designating this as `heavy` may reduce the coverage. Are the changes in this PR slowing down these 3d tests?

Reply:
Yes it runs slower and timeout during testing

Reply:
I wonder if we could do some profiling on it -- if you think that makes sense. Just wondering if these changes can significantly increase the runtime for bigger problems.

Reply:
This makes sense - we can at least check for this typical problem.

Reply:
Umm looks like the new implementation is slowing down the simulation quite a lot. Originally the test is taking ~3sec and the new implementation is making this to take ~40sec. About 60% of the time is spent in `ComputeMotarFunctor` while computing the Jacobian (40%) and residual (20%).

Reply:
So definitely more work to be done to improve the efficiency...

Reply:
I observed the same thing with this implementation (around 3x slowdown or so is what I said previously). Is most of the time being spent computing dual shape functions or is it something else?

Reply:
Oh btw, the `FEProblemBase::reinitElemFaceRef` and `FEProblemBase::reinitNeighborFaceRef` are the two under `ComputeMotarFunctor` that take the most time . Any ideas about how to improve this?

Ping @jbadger95

Reply:
Hmm I'm not sure xi1_pts and xi2_pts are being cleared during each loop iteration, might want to verify that those aren't growing

Reply:
@dewenyushu insert is just putting the new points at the beginning of the vector, that's my mistake I'm not sure what I was thinking there.

Reply:
I'm really surprised this didn't break more tests, it should be messing up all the 2D and 3D mortar integrals right? Sorry this was a major oversight, let me know if it fixes the cost and test tolerance issues!

Reply:
Sorry my email was not popping up new messages today for me somehow...

> Is most of the time being spent computing dual shape functions or is it something else?

For this test we do not have dual basis. The heavy part is in the `FEProblemBase::reinitElemFaceRef` and `FEProblemBase::reinitNeighborFaceRef`.

Good point on the xi1_pts and xi2_pts. Let me make that change and test it again.

> it should be messing up all the 2D and 3D mortar integrals right?

Maybe this is one of the origins of all the diffs in this PR. Let me check

Reply:
@jbadger95 You are right. The slowdown is due to the `xi1_pts` and `xi2_pts` not cleared for each msm element. The test now finishes below 3 sec. Glad you point this out!

The contact tests under `3d-mortar-contact` is still diffing - The displacements still needs a tolerance of 1e-3. The LMs requires a tolerance of 1e-4. I'll go ahead and regold these.

Reply:
Fantastic, @dewenyushu and @jbadger95! Can those small diffs be attributed to the slight changes in the location of quadrature points in the msm (IIRC)?

Reply:
@recuero yes I would think the diffs are coming from the quadrature changes

Reply:
@recuero It could be partially due to that but I'd guess it's due to the new computation of the dual shape functions. Remember now the dual shape functions are deformation dependent (we use the jacobian of the secondary element to compute the dual coeffs so for quad4 faces we should be getting different dual shape functions). @dewenyushu you could test that by setting the option to compute with the default quadrature rules.